src/io.rs

```diff
@@ -101,6 +118,91 @@ fn read_file_to_string(s: &str) -> IO<String> {
     }
 }
 
+fn read_file(file: &GluonFile, count: usize) -> IO<RuntimeResult<Option<Vec<u8>>, String>> {
+    if count == 0 {
```

Comment:
I don't think it is worth optimizing for this case, in practice it shouldn't happen (very often) making it a minor slow down in the common case and it also shadows any errors for zero-byte reads since it never touches the file. The standard library doesn't do the check either so better to follow suit.

Reply:
Yeah, it doesn't do anything helpful and being consistent makes more sense. Should the function even return an `Option` then? I guess it's not really necessary, but it is more convenient to use because you can pattern match on it.